### PR TITLE
Update Revues.org to OpenEdition Journals

### PR DESCRIPTION
**Pull request by jfriviere**
Revues.org Platform has changed its name to OpenEdition Journals.
Journals URL have changed form http://XXX.revues.org to https://journals.openedition.org/XXX

I updated the translator Revues.org.js as follow.  Test cases are successfull.
Thanks !
JFR

Translator updates :
- Rename file from Revues.org.js to OpenEdition Journals.js
- New urls processing
- Abstract and tags retrieved by the translator are now abstract and tags in document language (were previously in the browsing language used by the user)
- Update urls in translator and test cases
- Update test cases

### CODE REVIEW

**Comment by jfriviere** [comment]
I have fixed errors and warning reported by Travis.

**Comment by jfriviere** [comment]
Hi all, 
A kind reminder for this pull request. 
Is there anything I can do to complete the request ?
Thanks

File: OpenEdition Journals.js [hunk 1]
```diff
@@ -148,32 +163,36 @@ function doWeb(doc, url) {
...
-			var urls = new Array();
-			for (var i in selectedItems) {
+			var urls = [];
+			for (let i in selectedItems) {
 				urls.push(i);
 			}
 
-			ZU.processDocuments(urls, function(doc) {
-				scrape(doc, doc.location.href)
+			ZU.processDocuments(urls, function (doc) {
```
Comment (adam3smith) [inline OpenEdition Journals.js:174]: this should just work as `ZU.processDocuments(urls, scrape)` does it not?

File: OpenEdition Journals.js [hunk 2]
```diff
@@ -131,13 +147,12 @@ function scrape(doc, url) {
...
 	if (detectWeb(doc, url) == "multiple") {
-		var results = ZU.xpath(doc, '//div[@id="inside"]/div[@class="sommaire"]\
-			/dl[@class="documents"]/dd[@class="titre"]');
+		var results = ZU.xpath(doc, '//div[@id="inside"]/div[@class="sommaire"]/dl[@class="documents"]/dd[@class="titre"]');
 		if (!results.length) {
 			results = ZU.xpath(doc, '//ul[@class="summary"]//div[@class="title"]');
 		}
 
-/* From old code: When is this needed?
+		/* From old code: When is this needed?
```
Comment (adam3smith) [inline OpenEdition Journals.js:155]: delete this block then

File: OpenEdition Journals.js [hunk 3]
```diff
@@ -28,96 +48,92 @@ function detectWeb(doc, url) {
...
 		
-			//The site lists all editor of journals as editor in the header. Turn them into contributors. 
-			//I don't think there is a use case for editors for journal articles
-			if (item.itemType === "journalArticle"){
-				for (i in item.creators){
-					if (item.creators[i].creatorType === "editor"){
+			// The site lists all editor of journals as editor in the header. Turn them into contributors.
+			// I don't think there is a use case for editors for journal articles
+			if (item.itemType === "journalArticle") {
+				for (let i in item.creators) {
```
Comment (adam3smith) [inline OpenEdition Journals.js:123]: don't use `for ... in` to iterate through arrays. `for ... of` is preferred, explicitly increasing the index as you do elsewhere is also fine.

File: OpenEdition Journals.js [hunk 4]
```diff
@@ -28,96 +48,92 @@ function detectWeb(doc, url) {
...
+		|| url.match(/document\d+/)) {
 		return "journalArticle";
 	}
+	return false;
 }
 
 function scrape(doc, url) {
-	//is this still necessary??
-	if (url.match(/persee\-\d+/)) {
+	var translator;
```
Comment (adam3smith) [inline OpenEdition Journals.js:63]: This is better handled by using `let translator = ...`  in each individual condition for clarity

File: OpenEdition Journals.js [hunk 5]
```diff
@@ -28,96 +48,92 @@ function detectWeb(doc, url) {
...
-			/dl[@class="documents"]/dd[@class="titre"]/a').length ||
-		ZU.xpath(doc, '//ul[@class="summary"]//div[@class="title"]/a').length) {
+	if (ZU.xpath(doc, '//div[@id="inside"]/div[@class="sommaire"]/dl[@class="documents"]/dd[@class="titre"]/a').length
+		|| ZU.xpath(doc, '//ul[@class="summary"]//div[@class="title"]/a').length) {
 		return "multiple";
-	} else if (ZU.xpath(doc, '//h1[@id="docTitle"]/span[@class="text"]').length ||
-		url.match(/document\d+/)) {
+	}
+	else if (ZU.xpath(doc, '//h1[@id="docTitle"]/span[@class="text"]').length
+		|| url.match(/document\d+/)) {
```
Comment (adam3smith) [inline OpenEdition Journals.js:56]: avoid using `.match()` for testing regex. use either `/regex/.test(string)` or `string.search(regex) != -1`

File: OpenEdition Journals.js [hunk 6]
```diff
@@ -28,96 +48,92 @@ function detectWeb(doc, url) {
...
 	}
+	return false;
 }
 
 function scrape(doc, url) {
-	//is this still necessary??
-	if (url.match(/persee\-\d+/)) {
+	var translator;
+	// Example: https://journals.openedition.org/remi/persee-144614
+	if (url.match(/persee-\d+/)) {
```
Comment (adam3smith) [inline OpenEdition Journals.js:65]: same as above on `match()`

File: OpenEdition Journals.js (hunk 1 above)
Comment (jfriviere) [reply OpenEdition Journals.js]: Yes, it does

File: OpenEdition Journals.js (hunk 4 above)
Comment (jfriviere) [reply OpenEdition Journals.js]: It's done

File: OpenEdition Journals.js (hunk 5 above)
Comment (jfriviere) [reply OpenEdition Journals.js]: it's fixed

**Comment by jfriviere** [comment]
Thanks a lot for your review!
I've fixed all your comments and I rebased and squashed all the ESlint fixes in one commit. I hope I did the right way (not sure...)

**Comment by jfriviere** [comment]
Hi @adam3smith 
I have fixed all the comments you made in your review on July. Is there anything else I should do? If not, could you merge the pull request so that Zotero will work back on http://journals.openedition.org? 
Thanks for your valuable work!
Jean-François

**Comment by AbeJellinek** [comment]
Sorry this took so long to get merged!

**Comment by jfriviere** [comment]
Thanks for merging! It's still relevant!

**Pull request merged**